### PR TITLE
switch to alpine slim and do magic for build-essential

### PR DESCRIPTION
**Pull request by curcuz**
this produces a working build of around 100MB - a stunning improvement!

### CODE REVIEW

**Comment by shaunmulligan** [comment]
@curcuz how big is the final image? I think if you use the multi stage stuff like here https://github.com/resin-io-playground/nodejs-multistage-docker , you can probably get it to be about 70mb total

**Comment by curcuz** [comment]
@shaunmulligan it's 100MB and in theory in this specific case my approach should lead to same result as multi-stage, unless our alpine-node base images are bigger than alpine+multistage/build node :)

File: Dockerfile.template [hunk 1]
```diff
@@ -1,15 +1,17 @@
 # base-image for node on any machine using a template variable,
 # see more about dockerfile templates here:http://docs.resin.io/pages/deployment/docker-templates
 # Note the node:slim image doesn't have node-gyp
-FROM resin/%%RESIN_MACHINE_NAME%%-node:6
+FROM resin/%%RESIN_MACHINE_NAME%%-alpine-node:slim
```
Comment (Page-) [inline Dockerfile.template]: This should be 6-slim @curcuz, otherwise it'll track the latest node version and potentially break on new major node versions

Comment (curcuz) [inline Dockerfile.template]: @Page- thanks, addressed here: https://github.com/resin-io-projects/resin-node-red/pull/6

**Pull request merged**